### PR TITLE
Add support for MGMT_COMMISSIONER_GET and MGMT_COMMISSIONER_SET

### LINKED ISSUE #430: Implement meshcop commissioner dataset and related mgmt commands
Certificate test 9.2.1, 9.2.2 and 9.2.6 are used to verify different roles' behavior while receiving commissioner dataset mgmt commands.

Need to support for passing the certificate test.

### PR DESCRIPTION
**Pull request by luwannest**
- Implement MGMT_COMMISSIONER_GET and MGMT_COMMISSIONER_SET
- Add CLI for sending this two commands

These are needed by Certificate test cast 9.2.1 and 9.2.2

Resolves #430.

### CODE REVIEW

File: src/core/thread/network_data.cpp [hunk 1]
```diff
@@ -658,6 +658,136 @@ ThreadError NetworkData::SendServerDataNotification(uint16_t aRloc16)
     return error;
 }
 
+ThreadError NetworkData::SendGetCommissioningRequest(const uint8_t *aTlvs, uint8_t aLength)
```
Comment (jwhui) [inline src/core/thread/network_data.cpp]: Should this be a part of `MeshCoP::Commissioner`?

File: src/core/thread/network_data.cpp [hunk 2]
```diff
@@ -658,6 +658,136 @@ ThreadError NetworkData::SendServerDataNotification(uint16_t aRloc16)
...
+
+    if (error != kThreadError_None && message != NULL)
+    {
+        message->Free();
+    }
+
+    return error;
+}
+
+ThreadError NetworkData::SendSetCommissioningRequest(const otCommissioningDataset &aDataset,
```
Comment (jwhui) [inline src/core/thread/network_data.cpp]: Should this be a part of `MeshCoP::Commissioner`?

File: src/core/thread/network_data.cpp [hunk 3]
```diff
@@ -658,6 +658,136 @@ ThreadError NetworkData::SendServerDataNotification(uint16_t aRloc16)
...
+    SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
+
+    if (aDataset.mIsSteeringDataSet)
+    {
+        MeshCoP::SteeringDataTlv steeringData;
+        steeringData.Init();
+
+        for (uint8_t index = 0; index < OT_STEERING_DATA_MAX_LENGTH; index++)
+        {
+            for (uint8_t offset = 0; offset < 8; offset++)
```
Comment (jwhui) [inline src/core/thread/network_data.cpp]: Can we just `Append()` `mSteeringData.m8` directly onto the message?

**Comment by luwannest** [comment]
I am now working on 9.2.1 to 9.2.6 THCI and to see whether it can pass them or still need some fixes.

File: src/core/thread/network_data.cpp (hunk 1 above)
Comment (luwannest) [inline src/core/thread/network_data.cpp]: Yes, commisionser sends it, but commissioning data is part of network data. So I put it here.
I can move it.

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -1977,6 +1977,97 @@ void Interpreter::ProcessCommissioner(int argc, char *argv[])
...
+            }
+            else if (strcmp(argv[index], "sessionid") == 0)
+            {
+                tlvs[length++] = OT_MESHCOP_TLV_COMM_SESSION_ID;
+            }
+            else if (strcmp(argv[index], "steeringdata") == 0)
+            {
+                tlvs[length++] = OT_MESHCOP_TLV_STEERING_DATA;
+            }
+            else if (strcmp(argv[index], "joinerudpport") == 0)
```
Comment (jwhui) [inline src/cli/cli.cpp]: @turon, this contradicts your earlier suggestion of mapping to the spec.

File: src/core/thread/network_data.cpp [hunk 5]
```diff
@@ -658,6 +658,136 @@ ThreadError NetworkData::SendServerDataNotification(uint16_t aRloc16)
...
+
+    mSocket.Open(&HandleUdpReceive, this);
+
+    for (size_t i = 0; i < sizeof(mCoapToken); i++)
+    {
+        mCoapToken[i] = static_cast<uint8_t>(otPlatRandomGet());
+    }
+
+    header.Init();
+    header.SetVersion(1);
```
Comment (jwhui) [inline src/core/thread/network_data.cpp]: I'd make this a separate PR.

**Comment by luwannest** [comment]
This PR helps to pass 9.2.1 (on-mesh commissioner) and 9.2.2.

**Comment by luwannest** [comment]
I can see the AppVeyor build failed. But the log shows that network_data_leader.cpp line 611, potentially uninitialized local variable 'sessionId' used. But no sessionId is used there?

**Comment by jwhui** [comment]
FYI, the build failure was introduced in #641.  I just merged #650 which moves the failing test back into the XFAIL list.  You'll need to rebase on top of master to get the checks to pass.

**Comment by jwhui** [comment]
Regarding the AppVeyor build, it seems that `sessionId` does exist at [line 611](https://github.com/openthread/openthread/pull/634/files#diff-147b8306bdb58dd603ded796edecb2f9R611) in your PR.
